inst/include/cpp11/protect.hpp

```diff
@@ -54,20 +54,36 @@ inline void set_option(SEXP name, SEXP value) {
   SETCAR(opt, value);
 }
 
-inline Rboolean& get_should_unwind_protect() {
+inline Rboolean* setup_should_unwind_protect() {
   SEXP should_unwind_protect_sym = Rf_install("cpp11_should_unwind_protect");
   SEXP should_unwind_protect_sexp = Rf_GetOption1(should_unwind_protect_sym);
+
   if (should_unwind_protect_sexp == R_NilValue) {
+    // Allocate and initialize once, then let R manage it.
+    // That makes this a shared global across all compilation units.
     should_unwind_protect_sexp = PROTECT(Rf_allocVector(LGLSXP, 1));
+    SET_LOGICAL_ELT(should_unwind_protect_sexp, 0, TRUE);
     detail::set_option(should_unwind_protect_sym, should_unwind_protect_sexp);
     UNPROTECT(1);
   }
 
-  Rboolean* should_unwind_protect =
-      reinterpret_cast<Rboolean*>(LOGICAL(should_unwind_protect_sexp));
-  should_unwind_protect[0] = TRUE;
+  return reinterpret_cast<Rboolean*>(LOGICAL(should_unwind_protect_sexp));
+}
+
+inline Rboolean* access_should_unwind_protect() {
+  // Setup is run once per compilation unit, but all compilation units
+  // share the same global option, so each compilation unit's static pointer
+  // will point to the same object.
+  static Rboolean* p_should_unwind_protect = setup_should_unwind_protect();
```

Comment:
The function holding the static local variable is no longer templated, so each compilation unit will only have to run `setup_should_unwind_protect()` 1 time